docs/sources/panels-visualizations/visualizations/state-timeline/index.md

```diff
@@ -19,9 +19,70 @@ weight: 100
+
+<!-- video TBA here -->
+
+## Supported data formats
+
+The state timeline panel works best if you have data capturing the various states of entities over time, formatted as a table. The data must include:
+
+- **Timestamps** - Indicate when each state change occurred. This could also be the start time for the state change. You can also add an optional timestamp to indicate the end time for the state change.
+- **Entity name/identifier** - Represents the name of the entity you're trying to monitor.
+- **State value** - Represents the state value of the entity you're monitoring. These can be string, numerical, or boolean states.
+
+Each state ends when the next state begins or when there is a `null` value.
+
+### Examples
+
+The following tables are examples of the type of data you need for a state timeline visualization and how it should be formatted.
+
+#### Single time column with null values
+
+| Timestamps          | Server A | Server B |
+| ------------------- | -------- | -------- |
+| 2024-02-29 8:00:00  | Up       | Up       |
+| 2024-02-29 8:15:00  | null     | Up       |
+| 2024-02-29 8:30:00  | Down     | null     |
+| 2024-02-29 8:45:00  |          | Up       |
+| 2024-02-29 9:00:00  | Up       |          |
+| 2024-02-29 9:15:00  | Up       | Down     |
+| 2024-02-29 9:30:00  | Up       | Down     |
+| 2024-02-29 10:00:00 | Down     | Down     |
+| 2024-02-29 10:30:00 | Warning  | Down     |
+
+The data is converted as follows, with the [null and empty values visualized as gaps](https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/state-timeline/#connect-null-values) in the state timeline:
+
+{{< figure src="/static/img/docs/state-timeline-panel/state-timeline-with-null-values.png" max-width="1025px" alt="A state timeline panel with null values showing the status of two servers" >}}
+
+#### Two time columns without null values
+
+| Start time          | End time            | Server A | Server B |
+| ------------------- | ------------------- | -------- | -------- |
+| 2024-02-29 8:00:00  | 2024-02-29 8:15:00  | Up       | Up       |
+| 2024-02-29 8:15:00  | 2024-02-29 8:30:00  | Up       | Up       |
+| 2024-02-29 8:45:00  | 2024-02-29 9:00:00  | Down     | Up       |
+| 2024-02-29 9:00:00  | 2024-02-29 9:15:00  | Down     | Up       |
+| 2024-02-29 9:30:00  | 2024-02-29 10:00:00 | Down     | Down     |
+| 2024-02-29 10:00:00 | 2024-02-29 10:30:00 | Warning  | Down     |
+
+The data is converted as follows:
+
+{{< figure src="/static/img/docs/state-timeline-panel/state-timeline-with-two-timestamps.png" max-width="1025px" alt="A state timeline panel with two time columns showing the status of two servers" >}}
+
+If your query results aren't in a table format like the preceding examples, especially for time-series data, you can apply specific [transformations](https://stackoverflow.com/questions/68887416/grafana-state-timeline-panel-with-values-states-supplied-by-label) to achieve this.
```

Comment:
@imatwawana, I've updated the link to this community post instead, as this is very specific to state timeline transformations. Leon has kindly pointed out that saying just "transformations" is not that useful since this can be applied to other panel types, too.